packages/sommelier.rb

```diff
@@ -2,90 +2,197 @@
+  (( wait = wait - 1 ))
+sleep 1
+done
+SOMMPIDS=$(pgrep -f sommelier.elf 2> /dev/null)
+if [[ \"$(pgrep -fc sommelier.elf 2> /dev/null)\" == \"2\" ]]; then
+  echo \"sommelier processes running: $(echo $SOMMPIDS)\"
+else
+  echo \"sommelier failed to start\"
+  exit 1
+fi
+EOF"
+
+      # stopsommelier
+      system "cat <<'EOF'> stopsommelier
+#!/bin/bash
+SOMM=$(pgrep -fc sommelier.elf 2> /dev/null)
+if [[ \"$SOMM\" -gt \"0\" ]]; then
+  pkill -f sommelier.elf &>/dev/null
+else
+  exit 0
+fi
+if [[ \"$(pgrep -fc sommelier.elf 2> /dev/null)\" -gt \"0\" ]]; then
+  echo \"sommelier failed to stop\"
+  exit 1
+else
+  echo \"sommelier stopped\"
+fi
+EOF"
+
+      # restartsommelier
+      system "echo '#!/bin/bash' > restartsommelier"
+      system "echo 'stopsommelier && startsommelier' >> restartsommelier"
+      end
     end
   end
-
+    
   def self.install
-    Dir.chdir ("sommelier") do
-      system "make", "PREFIX=#{CREW_PREFIX}", "SYSCONFDIR=#{CREW_PREFIX}/etc", "DESTDIR=#{CREW_DEST_DIR}", "install"
-      system "install -Dm755 sommelierd #{CREW_DEST_PREFIX}/sbin/sommelierd"
-      system "install -Dm755 initsommelier #{CREW_DEST_PREFIX}/bin/initsommelier"
-      system "install -Dm755 stopsommelier #{CREW_DEST_PREFIX}/bin/stopsommelier"
-      system "install -Dm755 startsommelier #{CREW_DEST_PREFIX}/bin/startsommelier"
-      system "install -Dm644 .sommelier.env #{CREW_DEST_HOME}/.sommelier.env"
+    Dir.chdir ("vm_tools/sommelier") do
+      system "DESTDIR=#{CREW_DEST_DIR} ninja -C build install"
+      Dir.chdir ("build") do
+        FileUtils.mv "#{CREW_DEST_PREFIX}/bin/sommelier", "#{CREW_DEST_PREFIX}/bin/sommelier.elf"
+        system "install -Dm755 sommelier_sh #{CREW_DEST_PREFIX}/bin/sommelier"
+        system "install -Dm755 sommelierd #{CREW_DEST_PREFIX}/sbin/sommelierd"
+        system "install -Dm755 startsommelier #{CREW_DEST_PREFIX}/bin/startsommelier"
+        FileUtils.ln_sf 'startsommelier', "#{CREW_DEST_PREFIX}/bin/initsommelier"
+        system "install -Dm755 stopsommelier #{CREW_DEST_PREFIX}/bin/stopsommelier"
+        system "install -Dm755 restartsommelier #{CREW_DEST_PREFIX}/bin/restartsommelier"
+        system "install -Dm755 sommelierrc #{CREW_DEST_PREFIX}/etc/sommelierrc"
+        system "install -Dm644 .sommelier/CLUTTER_BACKEND #{CREW_DEST_HOME}/.sommelier/CLUTTER_BACKEND"
+        system "install -Dm644 .sommelier/DISPLAY #{CREW_DEST_HOME}/.sommelier/DISPLAY"
+        system "install -Dm644 .sommelier/GDK_BACKEND #{CREW_DEST_HOME}/.sommelier/GDK_BACKEND"
+        system "install -Dm644 .sommelier/MESA_LOADER_DRIVER_OVERRIDE #{CREW_DEST_HOME}/.sommelier/MESA_LOADER_DRIVER_OVERRIDE"
+        system "install -Dm644 .sommelier/SCALE #{CREW_DEST_HOME}/.sommelier/SCALE"
+        system "install -Dm644 .sommelier/WAYLAND_DISPLAY #{CREW_DEST_HOME}/.sommelier/WAYLAND_DISPLAY"
+        system "install -Dm644 .sommelier/XDG_RUNTIME_DIR #{CREW_DEST_HOME}/.sommelier/XDG_RUNTIME_DIR"
```

Comment:
Can't we include all these environment variables in one file?  Just thinking about simplicity.

Reply:
Rationale from error message I see otherwise:

https://chromium.googlesource.com/chromiumos/docs/+/master/security/noexec_shell_scripts.md

Reply:
Would be smart to simply source a file with the variables, as that wouldn't cause any issues. Just source the file from `startsommelier` and export the vars.

Reply:
Now sourcing from ~/.sommelier.env again.